File: 6KYU/Pair of gloves/solution.cs
using System.Linq;
using System;

public class Kata {
  public static int NumberOfPairs(string[] gloves) {
    var grps = gloves.GroupBy(x => x);
    var pairs = 0;
    foreach(var g in grps)
    {
      pairs += g.Count() / 2;
    }
      
    return pairs;
  }
}

File: 6KYU/Pair of gloves/solution.go
package kata

func NumberOfPairs(gloves []string) int {
	glovesColours := make(map[string]int)

	for _, glove := range gloves {
		glovesColours[glove] = glovesColours[glove] + 1
	}

	var count int = 0

	for _, v := range glovesColours {
		count = count + (v / 2)
	}

	return count
}
